RevenueCatUI/CustomerCenter/ViewModels/PromotionalOfferViewModel.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PromotionalOfferViewModel.swift
+//
+//
+//  Created by Cesar de la Vega on 17/6/24.
+//
+
+import Foundation
+import RevenueCat
+
+#if os(iOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+@MainActor
+class PromotionalOfferViewModel: ObservableObject {
+
+    private(set) var localization: CustomerCenterConfigData.Localization?
+
+    @Published
+    private(set) var promotionalOfferData: PromotionalOfferData?
+    @Published
+    private(set) var transaction: StoreTransaction?
+    @Published
```

Comment:
Do we need these to be `@Published`?

Reply:
do you mean `error`?

Reply:
yes, not used either of them. I think I am going to leave error though, and dismiss the promo offer sheet for now if there's an error